ts/defs.bzl

```diff
@@ -294,7 +294,6 @@ def ts_project(
 
         # These options are always passed on the tsc command line so don't include them
         # in the tsconfig. At best they're redundant, but at worst we'll have a conflict
-        out_dir = compiler_options.pop("outDir", out_dir)
```

Comment:
I think we would still want this so we can do `tsconfig = {... outDir ...}` without also needing `ts_project(tsconfig)`? Just like thigns like `declaration`, so maybe just move it up a few lines?

Reply:
Then in many places where you added `ts_project(tsconfig = {... outDir ... })` we can drop the `ts_project(out_dir)`?